### PR TITLE
Media types for plain json schema, JWT, and SD-JWT

### LINKED ISSUE #157: Add note about media types
Based on a slack conversation regarding media types for JSON Schema.

We should recommend:

```
application/schema+json
```

Or

```
application/schema+json; schema="http://json-schema.org/draft-07/schema#
```

When schemas are dereferenced via http (cover both accept and content-type headers).

Related comments:

- https://github.com/orgs/json-schema-org/discussions/198
- https://github.com/ietf-wg-httpapi/mediatypes/blob/main/draft-ietf-httpapi-rest-api-mediatypes.md#json-schema-media-types
- https://json-schema.org/draft/2020-12/json-schema-core.html#name-iana-considerations

### PR DESCRIPTION
**Pull request by decentralgabe**
fix #157

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-json-schema/pull/198.html" title="Last updated on Aug 14, 2023, 5:43 PM UTC (ec30c54)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-json-schema/198/ab0a7f3...ec30c54.html" title="Last updated on Aug 14, 2023, 5:43 PM UTC (ec30c54)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1099,5 +1098,110 @@ <h2>Internationalization Considerations</h2>
...
+               <h2><code>application/schema+json</code></h2>
+               <p>
+                  This specification acknowledges the use of the <code>application/schema+json</code>
+                  Media Type specifically for identifying a [[JSON-SCHEMA]] with usage of the
+                  <a href="#jsonschema">JsonSchema</a> type, as registered in 
+                  <a href="#json-schema-specifications">JSON Schema specifications</a>.
+               </p>
+            </section>  
+            <section id="vc-jwt-schema-json-media-type">
+               <h2><code>application/vc+ld+jwt+schema+json</code></h2>
```
Comment (OR13) [inline index.html:1115]: I don't think you need this... in the case you are dereferencing a "JsonSchemaCredential" the response type will be:

1. `application/vc+ld+json` in the case of data integrity proof.
2. `application/vc+ld+json+jwt` or `application/vc+ld+json+sd-jwt` in the case of vc-jose-cose

File: index.html [hunk 2]
```diff
@@ -1099,5 +1098,110 @@ <h2>Internationalization Considerations</h2>
...
+                   </td>
+                 </tr>
+                 <tr>
+                   <td>Contact: </td>
+                   <td>
+                     W3C Verifiable Credentials Working Group <a href="mailto:public-vc-wg@w3.org">public-vc-wg@w3.org</a>
+                   </td>
+                 </tr>
+               </table>
+            </section>
```
Comment (OR13) [inline index.html:1203]:
```suggestion
```

**Review by OR13 — CHANGES_REQUESTED** [review]
Adjust PR to acknowledge the response type for dereferencing... expecting to see at least the following media types acknowledged, but none of them will be registered by this spec.

For type: "JsonSchema": 

- `application/schema+json` or `application/json`
- `application/openapi+yaml` (maybe) or `application/yaml`  (maybe)

For type: "JsonSchemaCredential": 

- `application/vc+ld+json` 
- `application/vc+ld+json+jwt` 
- `application/vc+ld+json+sd-jwt`

File: index.html (hunk 2 above)
Comment (decentralgabe) [reply index.html]: removed

File: index.html [hunk 3]
```diff
@@ -570,121 +570,141 @@ <h3>Evaluation</h3>
...
-            "description": "EmailCredential using JsonSchema",
-            "type": "object",
-            "properties": {
-              "credentialSubject": {
-                "type": "object",
-                "properties": {
-                  "emailAddress": {
-                    "type": "string",
-                    "format": "email"
+          <section class="normative">
```
Comment (decentralgabe) [inline index.html:573]: just updated formatting here...

File: index.html [hunk 4]
```diff
@@ -1099,5 +1118,34 @@ <h2>Internationalization Considerations</h2>
...
+                  or <code>application/yaml</code> may be used.
+               </p>
+            </section>  
+            <section class="normative" id="media-types-for-jsonschemacredential">
+               <h2><code>JsonSchemaCredential</code></h2>
+               <p>
+                  This specification acknowledges the use of the <code>application/vc+ld+json</code>,
+                  <code>application/vc+ld+json+jwt</code>, and <code>application/vc+ld+json+sd-jwt</code> Media Types
+                  specifically for identifying a [[JSON-SCHEMA]] with usage of the <a href="#jsonschemacredential">
+                  JsonSchemaCredential</a> type, as registered in the [[VC-DATA-MODEL-2]] and [[VC-JWT]] specifications.
```
Comment (andresuribe87) [inline index.html:1145]:
```suggestion
                  JsonSchemaCredential</a> type, as registered in the [[VC-DATA-MODEL-2]] and [[VC-JOSE-COSE]] specifications.
```

File: index.html [hunk 5]
```diff
@@ -1099,5 +1118,34 @@ <h2>Internationalization Considerations</h2>
...
+                  When using the <a href="#jsonschema">JsonSchema</a> type with a YAML
+                  representation of a JSON Schema, defined by [[OAS]], the types <code>application/openapi+yaml</code>
+                  or <code>application/yaml</code> may be used.
+               </p>
+            </section>  
+            <section class="normative" id="media-types-for-jsonschemacredential">
+               <h2><code>JsonSchemaCredential</code></h2>
+               <p>
+                  This specification acknowledges the use of the <code>application/vc+ld+json</code>,
+                  <code>application/vc+ld+json+jwt</code>, and <code>application/vc+ld+json+sd-jwt</code> Media Types
```
Comment (andresuribe87) [inline index.html:1143]: `application/vc+ld+json+sd-jwt` is registered in https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-05.html#section-12.2

I'm not sure how to reference that spec though.

File: index.html [hunk 6]
```diff
@@ -1099,5 +1118,34 @@ <h2>Internationalization Considerations</h2>
...
+                  specifically for identifying a [[JSON-SCHEMA]] with usage of the <a href="#jsonschemacredential">
+                  JsonSchemaCredential</a> type, as registered in the [[VC-DATA-MODEL-2]] and [[VC-JOSE-COSE]] specifications.
```
Comment (decentralgabe) [inline index.html:1145]:
```suggestion
                  JsonSchemaCredential</a> type, as registered in the [[VC-DATA-MODEL-2]], [[VC-JOSE-COSE]], and [[SD-JWT]] specifications respectively.
```

**Review by andresuribe87 — APPROVED** [review]

File: index.html [hunk 7]
```diff
@@ -694,11 +714,11 @@ <h2>Implementation Considerations</h2>
...
         <section class="informative">
-          <h3>Validation of Specific Properties in a Credential</h3>
+          <h3>Credential Property Validation</h3>
           <p>
-            Implementers may wish to validate certain sections of a <a>verifiable credential</a>.
-            To do this, <a>credential schemas</a> can be constructed to specify application
-            to subsets of a given credential.
+            Implementers may wish to validate certain sections or properties of a 
+            <a>verifiable credential</a>. To do this, <a>credential schemas</a> can be 
+            constructed to specify application to subsets of a given credential.
```
Comment (OR13) [inline index.html:721]: wording a bit awk

**Review by OR13 — APPROVED** [review]

File: index.html [hunk 8]
```diff
@@ -694,11 +714,11 @@ <h2>Implementation Considerations</h2>
           may consider.
         </p>
         <section class="informative">
-          <h3>Validation of Specific Properties in a Credential</h3>
+          <h3>Credential Property Validation</h3>
           <p>
-            Implementers may wish to validate certain sections of a <a>verifiable credential</a>.
-            To do this, <a>credential schemas</a> can be constructed to specify application
-            to subsets of a given credential.
+            Implementers may wish to validate certain sections or properties of a
```
Comment (decentralgabe) [inline index.html:719]:
```suggestion
            Implementers may wish to validate certain properties in a 
```

File: index.html [hunk 9]
```diff
@@ -694,11 +714,11 @@ <h2>Implementation Considerations</h2>
...
         <section class="informative">
-          <h3>Validation of Specific Properties in a Credential</h3>
+          <h3>Credential Property Validation</h3>
           <p>
-            Implementers may wish to validate certain sections of a <a>verifiable credential</a>.
-            To do this, <a>credential schemas</a> can be constructed to specify application
-            to subsets of a given credential.
+            Implementers may wish to validate certain properties in a 
+            <a>verifiable credential</a>. To do this, <a>credential schemas</a> can be 
+            constructed to specify application to subsets of a given credential.
```
Comment (decentralgabe) [inline index.html:721]:
```suggestion
            constructed to validate subsets of a credential's data.
```

File: index.html (hunk 7 above)
Comment (decentralgabe) [reply index.html]: updated

File: index.html [hunk 10]
```diff
@@ -570,121 +570,141 @@ <h3>Evaluation</h3>
...
-                    "format": "email"
+               }
+               </pre>
+            </p>
+            <p>
+              Validation according to the spec [[JSON-SCHEMA-2020-12]] yields a <i>Success</i> result
+              when applying it to the VC below.
+            </p>
+            <p>
+               <pre class="example vc" title="Example Verifiable Credential - validations Success">
```
Comment (decentralgabe) [inline index.html:603]:
```suggestion
               <pre class="example vc-jwt" title="Example Verifiable Credential - validations Success">
```

**Pull request merged**